src/zarr/core/group.py

```diff
@@ -149,15 +234,26 @@ async def open(
         cls,
         store: StoreLike,
         zarr_format: Literal[2, 3, None] = 3,
+        open_consolidated: bool = False,
```

Comment:
I'm not sure about this API yet. I only added it for Zarr V2. I think the main question to answer is:

What top-level behavior do we want if a user does `zarr.open_consolidated` on a hierarchy that doesn't have consolidated metadata (v2 or v3)? Silently not use consolidated metadata, or error loudly?

Reply:
The behavior is documented as

1. `open` uses consolidated metadata if present, doesn't if not.
2. Users can set `use_consolidated=True` to error if it's not found, `False` to bypass it if it's not present.
3. `open_consolidated` sets `use_consolidated=True` which will error if consolidated metadata is not found.